libsolutil/CommonIO.h

```diff
@@ -48,6 +48,46 @@ inline std::ostream& operator<<(std::ostream& os, bytes const& _bytes)
 namespace util
 {
 
+namespace detail
+{
+
+template <typename Predicate>
+struct RecursiveFileCollector
+{
+	Predicate predicate;
+	std::vector<boost::filesystem::path> result {};
+
+	RecursiveFileCollector& operator()(boost::filesystem::path const& _directory)
+	{
+		if (!boost::filesystem::is_directory(_directory))
+			return *this;
+		auto iterator = boost::filesystem::directory_iterator(_directory);
+		auto const iteratorEnd = boost::filesystem::directory_iterator();
+
+		while (iterator != iteratorEnd)
+		{
+			if (boost::filesystem::is_directory(iterator->status()))
+				(*this)(iterator->path());
```

Comment:
You already have a check against it being a dir at the beginning of the function. This one seems superfluous and, given how heavy disk operations typically are, removing it should save more than we lose on an extra function call.
```suggestion
			(*this)(iterator->path());
```